app/src/main/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragment.kt

```diff
@@ -20,41 +34,266 @@ class ScannerFragment : Fragment() {
+    private var imageCapture: ImageCapture? = null
+
+    private var _takingPicture: Boolean = false
+    val isTakingPicture get() = _takingPicture
+
+
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        val scannerViewModel = ViewModelProvider(this)[ScannerViewModel::class.java]
-
+        scannerViewModel = ViewModelProvider(this)[ScannerViewModel::class.java]
         _binding = FragmentScannerBinding.inflate(inflater, container, false)
         val root: View = binding.root
 
-        scannerViewModel.textInformationId.observe(viewLifecycleOwner) {
-            binding.scannerInformationText.text = getString(it)
+        // bind the view model to the layout when text or object is detected in image
+        scannerViewModel.textDetected.observe(viewLifecycleOwner) {
+            val toDisplay = StringBuilder()
+            toDisplay.append(MagicCardMatcher.match(it).toString())
+            toDisplay.append(it.text)
+
+            binding.scannerTextInImageText.text = toDisplay.toString()
+        }
+
+        binding.scannerSaveButton.setOnClickListener {
+            savePicture()
         }
 
-        val registerImage =
-            registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
-                scannerViewModel.setTextInformation(it.resultCode)
-                binding.scannerImage.setImageBitmap(
-                    Compatibility.getDataActivityResult(
-                        it,
-                        "data",
-                        Bitmap::class.java
-                    )
+        // Code adapted from https://developer.android.com/training/camerax
+        // Request camera permissions
+        if (allPermissionsGranted()) {
+            startCamera()
+        } else {
+            this.activity?.let {
+                ActivityCompat.requestPermissions(
+                    it,
+                    REQUIRED_PERMISSIONS,
+                    REQUEST_CODE_PERMISSIONS
                 )
             }
-
-        binding.scannerScanButton.setOnClickListener {
-            registerImage.launch(Intent(MediaStore.ACTION_IMAGE_CAPTURE))
         }
 
+        cameraExecutor = Executors.newSingleThreadExecutor()
+
         return root
     }
 
+
+    // Code adapted from https://developer.android.com/training/camerax
+    /**
+     * Check if all permissions are granted or not
+     */
+    private fun allPermissionsGranted() = REQUIRED_PERMISSIONS.all {
+        val baseContext = this.context ?: return false
+        ContextCompat.checkSelfPermission(baseContext, it) == PackageManager.PERMISSION_GRANTED
+    }
+
+    /**
+     * Check whether permissions have been granted after the request,
+     * replaces the onRequestPermissionsResult method from the activity
+     */
+    fun requestPermissionsResult(requestCode: Int, perm: Array<String>, results: IntArray) {
```

Comment:
"perm" and "results" parameters seems to be useless in this function

Reply:
yes, they are not used here, but they match the original prototype for handling the permissionsResults

Reply:
Ok, then I guess we can let it that way